proposals/z-index.md

```diff
@@ -0,0 +1,58 @@
+# z-index Convention
+
+## Problem
+
+We have no convention for choosing z-index. This means that developers can set an arbitrary z-index on a component that may have the side effect that it displays above or below another component of which it has no knowledge.
+
+## Solution
+
+### Base layers
+
+First we define layers in an object. These base layers are 10 apart so there is room to define hierarchies within layers when necessary.
+
+```js
+const layers = {
+  '-1': -1,
+  content: 10,
+  header: 20, // this includes, for example, the popover
+  modal: 30
+};
+```
+
+### Utility function
+
+We create a function that takes a layer key and returns the corresponding z-index value for that base layer.
+
+```js
+export function getZIndex(layer) {
+  try {
+    const value = layers[layer];
+
+    // throw an error if the key isn't defined in the layers object
+    if (!value) {
```

Comment:
We should use `if (!layer in layers)` because otherwise, currently this would error if the value was `0`.

Reply:
✅